cmd/keytransparency-monitor/main.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	mopb "github.com/google/keytransparency/impl/proto/monitor_v1_service"
+	mupb "github.com/google/keytransparency/impl/proto/mutation_v1_service"
+
+	"github.com/golang/glog"
+	"github.com/google/keytransparency/impl/monitor"
+	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/reflection"
+)
+
+var (
+	addr      = flag.String("addr", ":8099", "The ip:port combination to listen on")
+	vrfPath   = flag.String("vrf", "genfiles/vrf-pubkey.pem", "Path to VRF public key")
+	logPubKey = flag.String("log-key", "genfiles/trillian-log.pem", "Path to the trillian log's public key")
+	keyFile   = flag.String("key", "genfiles/server.key", "TLS private key file")
+	certFile  = flag.String("cert", "genfiles/server.pem", "TLS cert file")
+
+	pollPeriod = flag.Duration("poll-period", time.Second*5, "Maximum time between polling the key-server. Ideally, this is equal to the min-period of paramerter of the keyserver.")
+	ktURL      = flag.String("kt-url", "localhost:8080", "URL of key-server.")
+	ktPEM      = flag.String("kt-key", "genfiles/server.crt", "Path to kt-server's public key")
+	// TODO(ismail): are the IDs actually needed for verification operations?
+	mapID = flag.Int64("map-id", 0, "Trillian map ID")
```

Comment:
`mapID` is currently needed by the [verifier](https://github.com/google/keytransparency/blob/142a834769e003a154841f760511550e4c25fd2c/core/tree/sparse/verifier/verifier.go#L41). I've added a TODO to remove it as soon as the verifier is updated (similar to https://github.com/google/keytransparency/pull/708)